java/core/src/java/org/apache/orc/TimestampColumnStatistics.java

```diff
@@ -47,4 +47,19 @@
    * @return maximum value in UTC
    */
   Timestamp getMaximumUTC();
+
+  /**
+   * Get the lowest digits of the minimum value for the column.
+   * The minimum value is truncated at ms scale.
+   * @return the 6 lowest digits of minimum value
+   */
+  int getMinimumNanos();
```

Comment:
java.sql.Timestamp does support nanosecond. Is it better to use existing APIs and handle nanosecond transparently?

Reply:
We could do that, however I believe its more clear to handle the nanoseconds as separate methods to show how ns precision is handled in old Proto (default values) and new Proto with existing values.